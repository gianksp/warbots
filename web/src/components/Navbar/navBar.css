/*Custom CSS overwriting default imported CSS goes here*/

.topPadding {
    padding-top: 8px!important;
}