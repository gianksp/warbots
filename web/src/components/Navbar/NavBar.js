import React from 'react'
import PropTypes from 'prop-types'
import { NavLink } from 'react-router-dom'
import 'bulma/css/bulma.css'
import './navBar.css'
export const NavBar = (props) => {
    return (
     <div className="column is-half is-offset-one-quarter">
         <nav className="navbar ">
             <div id="navMenuExample" className="navbar-menu">
                 <div className="navbar-start">
                     <NavLink exact activeClassName="navbar-item" to='/'>{props.menu1}</NavLink>
                     <NavLink className="navbar-item topPadding" to='/about'>{props.menu2}</NavLink>
                 </div>
             </div>
         </nav>
     </div>					
    )
}
NavBar.propTypes = {
    menu1: PropTypes.string,
    menu2: PropTypes.string
}
