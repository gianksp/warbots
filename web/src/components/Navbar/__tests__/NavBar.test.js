import React from 'react'
import renderer from 'react-test-renderer'
import { NavBar } from './components/Navbar/NavBar'
import { MemoryRouter } from 'react-router'


describe('Navbar Component renders nav correctly', () => {
    it('renders nav elements correctly', () => {
        const rendered = renderer.create(
            <MemoryRouter>
			<NavBar/>
			</MemoryRouter>
        ).toJSON()
        expect(rendered).toMatchSnapshot()
    })
})