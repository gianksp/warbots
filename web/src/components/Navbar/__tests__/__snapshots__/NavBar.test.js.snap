// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar Component renders nav correctly renders nav elements correctly 1`] = `
<div>
  <nav
    className="navbar "
  >
    <div
      className="navbar-menu"
      id="navMenuExample"
    >
      <div
        className="navbar-start"
      >
        <a
          className="navbar-item"
          href="/"
          onClick={[Function]}
          style={Object {}}
        />
        <a
          className="navbar-item topPadding"
          href="/about"
          onClick={[Function]}
          style={undefined}
        />
      </div>
    </div>
  </nav>
</div>
`;
